openassessment/fileupload/backends/swift.py

```diff
@@ -63,6 +63,10 @@ def get_download_url(self, key):
             )
             raise FileUploadInternalError(ex)
 
+    def remove_file(self, key):
+        """
+        @TODO Need to be implemented to remove previously uploaded files
+        """
```

Comment:
Mmm. Ok. But it this case all instances that uses swift storage will be unworkable. `remove` method is used in the new `remove_all_uploaded_files` `XBlock.json_handler`:
```
    @XBlock.json_handler
    def remove_all_uploaded_files(self, data, suffix=''):  # pylint: disable=unused-argument
        """
        Removes all uploaded user files.

        """
        removed_num = 0
        for i in range(self.MAX_FILES_COUNT):
            removed = file_upload_api.remove_file(self._get_student_item_key(i))
            if removed:
                removed_num += 1
            else:
                break
        return {'success': True, 'removed_num': removed_num}
```
But this API is called every time before each upload.